src/hooks/useSetMobileDevice.ts

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectMobileDeviceState, setMobileDevice } from '../modules';
+
+export const useSetMobileDevice = () => {
+    const isMobileDevice = useSelector(selectMobileDeviceState);
+    const dispatch = useDispatch();
+
+    React.useEffect(() => {
+        const handleResize = () => {
+            const isMobileDeviceCurrent = (window.innerWidth < 768) || (window.innerHeight < 600);
+
+            window.console.log('here');
+
+            if (isMobileDevice !== isMobileDeviceCurrent) {
+                dispatch(setMobileDevice(isMobileDeviceCurrent));
+            }
+        };
+
+        window.addEventListener('resize', handleResize);
+        handleResize();
+
+        return () => window.removeEventListener('resize', handleResize);
+    }, [dispatch, isMobileDevice]);
+};
```

Comment:
What should be here:

    export const useSetMobileDevice = () => {
        const [isMobile, setIsMobile] = useState(window.innerWidth < 768) || (window.innerHeight < 600);

        React.useEffect(() => {
            const handleResize = () => {
                 setIsMobile(window.innerWidth < 768) || (window.innerHeight < 600);            
            };
            window.addEventListener('resize', handleResize);
            return () => window.removeEventListener('resize', handleResize);
        }, []);
    };

Reply:
we need to store it in the redux, not in the state